hamster_gtk/overview/widgets/charts.py

```diff
@@ -34,65 +38,77 @@ class Charts(Gtk.Grid):
     def __init__(self, totals):
         """Initialize widget."""
         super(Charts, self).__init__()
-        self.attach(self._get_category_widget(totals.category), 0, 0, 1, 1)
-        # [FIXME]
-        # Add analogous widgets for tags and activities?
+        self.set_column_spacing(20)
+        self.attach(Gtk.Label('Categories'), 0, 0, 1, 1)
+        self.attach(self._get_barcharts(totals.category), 0, 1, 1, 1)
+        self.attach(Gtk.Label('Activities'), 1, 0, 1, 1)
+        self.attach(self._get_barcharts(totals.activity), 1, 1, 1, 1)
+        self.attach(Gtk.Label('Dates'), 2, 0, 1, 1)
+        self.attach(self._get_barcharts(totals.date), 2, 1, 1, 1)
 
-    def _get_category_widget(self, category_totals):
-        """Return a widget to represent all categories in a column."""
-        grid = Gtk.Grid()
-        row = 0
-        # the highest amount of time spend in a category. This is the scale for
-        # all other totals.
-        max_total = max(category_totals.values())
-
-        # Build individual category 'rows'.
-        for category, total in category_totals.items():
-            category_label = Gtk.Label(category)
-            bar_chart = HorizontalBarChart(total.total_seconds(), max_total.total_seconds())
-            total_label = Gtk.Label(self._get_delta_string(total))
-            grid.attach(category_label, 0, row, 1, 1)
-            grid.attach(bar_chart, 1, row, 1, 1)
-            grid.attach(total_label, 2, row, 1, 1)
-            row += 1
-        return grid
+        self.override_background_color(Gtk.StateType.NORMAL, Gdk.RGBA(1, 1, 1, 1))
```

Comment:
That looks slightly better but still difficult to read. How do other applications deal with this?

![screenshot from 2017-06-30 19-35-42](https://user-images.githubusercontent.com/705123/27747537-ded87836-5dcb-11e7-89ce-6546678b9f2c.png)